src/app/components/list/lists.controller.js

```diff
@@ -81,22 +81,32 @@
       });
     }
 
-    var queryCallback = function (resp) {
-      $scope.totalItems = resp.headers["x-total-count"];
-      formatTypes(resp);
+    var queryCallback = function (lists, headers) {
+      $scope.totalItems = headers()["x-total-count"];
+      formatTypes(lists);
       $scope.listsLoaded = true;
-
-      resp.$httpPromise.then(function (lists) {
-        formatTypes(lists);
-      });
     };
 
     ListDataService.subscribe($scope, function () {
       $scope.currentPage = 1;
       $scope.pageChanged();
     });
 
-    $scope.lists = List.query($scope.request, queryCallback);
+    $scope.lists = List.query($scope.request, queryCallback, function (resp) {
```

Comment:
Might be better to handle this in the listData service, so the controller just asks for the lists, and the service handles whether to get it from the api or localforage. If we could make it consistent for users as well that'd be good.